# Artificial Neural Network - ANN

## Importing Libraries

In [1]:
import numpy as np
import pandas as pd
import tensorflow as tf

## Data Pre-processing

In [2]:
dataset = pd.read_excel('Folds5x2_pp.xlsx')

In [3]:
dataset.head()

AT      V       AP     RH      PE
0  14.96  41.76  1024.07  73.17  463.26
1  25.18  62.96  1020.04  59.08  444.37
2   5.11  39.40  1012.16  92.14  488.56
3  20.86  57.32  1010.24  76.64  446.48
4  10.82  37.50  1009.23  96.62  473.90

In [4]:
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, -1].values

In [5]:
print(X)

[[  14.96   41.76 1024.07   73.17]
 [  25.18   62.96 1020.04   59.08]
 [   5.11   39.4  1012.16   92.14]
 ...
 [  31.32   74.33 1012.92   36.48]
 [  24.48   69.45 1013.86   62.39]
 [  21.6    62.52 1017.23   67.87]]


In [6]:
print(y)

[463.26 444.37 488.56 ... 429.57 435.74 453.28]


In [8]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)

## Building ANN

In [11]:
ann = tf.keras.models.Sequential()

In [12]:
ann.add(tf.keras.layers.Dense(units=6, activation='relu'))

In [13]:
ann.add(tf.keras.layers.Dense(units=6, activation='relu'))

In [14]:
ann.add(tf.keras.layers.Dense(units=1))

## Training ANN

In [15]:
ann.compile(optimizer='adam', loss='mean_squared_error')

In [16]:
ann.fit(X_train,y_train,batch_size=32,epochs=100)

Epoch 1/100
240/240 [==============================] - 1s 2ms/step - loss: 29896.2480
Epoch 2/100
240/240 [==============================] - 0s 2ms/step - loss: 444.6917
Epoch 3/100
240/240 [==============================] - 0s 2ms/step - loss: 426.0401
Epoch 4/100
240/240 [==============================] - 0s 2ms/step - loss: 403.7910
Epoch 5/100
240/240 [==============================] - 0s 2ms/step - loss: 378.0683
Epoch 6/100
240/240 [==============================] - 0s 2ms/step - loss: 350.0031
Epoch 7/100
240/240 [==============================] - 0s 2ms/step - loss: 320.4482
Epoch 8/100
240/240 [==============================] - 0s 2ms/step - loss: 289.9391
Epoch 9/100
240/240 [==============================] - 0s 2ms/step - loss: 259.3388
Epoch 10/100
240/240 [==============================] - 0s 2ms/step - loss: 228.7363
Epoch 11/100
240/240 [==============================] - 0s 2ms/step - loss: 199.5868
Epoch 12/100
240/240 [==============================] - 0s 2ms/step - lo

## Predicting Results

In [17]:
y_pred = ann.predict(X_test)

60/60 [==============================] - 0s 1ms/step


In [20]:
np.set_printoptions(precision=2)
print(np.concatenate((y_pred.reshape(len(y_pred),1), y_test.reshape(len(y_test),1)),1))

[[430.89 431.23]
 [461.87 460.01]
 [465.33 461.14]
 ...
 [472.6  473.26]
 [439.43 438.  ]
 [458.62 463.28]]


In [21]:
from sklearn.metrics import r2_score
from sklearn.metrics import mean_absolute_error
r_squared = r2_score(y_test, y_pred)
mae = mean_absolute_error(y_test, y_pred)

In [22]:
print('r_square_value :',r_squared)
print("Mean Absolute Error (MAE):", mae)

r_square_value : 0.9153293730643752
Mean Absolute Error (MAE): 3.9399300468229574


R-squared is 0.915, indicating that approximately 91.5% of the variance in the target variable is explained by the model.

MAE of 3.94 means, on average, model's predictions are off by approximately 3.94 units from the actual values.